File: app/src/main/java/com/commonpepper/photosen/ui/activities/IntroActivity.java
package com.commonpepper.photosen.ui.activities;

import android.content.Intent;
import android.os.Bundle;
import android.util.TypedValue;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.commonpepper.photosen.R;
import com.commonpepper.photosen.ui.fragments.IntroSupportFragment;
import com.github.paolorotolo.appintro.AppIntro;
import com.github.paolorotolo.appintro.AppIntroFragment;
import com.github.paolorotolo.appintro.model.SliderPage;

public class IntroActivity extends AppIntro {
    public static final String SCROLL_TO_TOP = "SCROLL_TO_TOP";

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setGoBackLock(true);
        setDepthAnimation();

        TypedValue outValue = new TypedValue();
        getTheme().resolveAttribute(R.attr.selectableItemBackground, outValue, true);
        skipButton.setBackgroundResource(outValue.resourceId);
        doneButton.setBackgroundResource(outValue.resourceId);

        SliderPage page1 = new SliderPage();
        page1.setTitle(getString(R.string.welcome));
        page1.setDescription(getString(R.string.welcome_desc));
        page1.setImageDrawable(R.drawable.big_ic_launcher);
        page1.setBgColor(R.color.colorAccent);
        addSlide(AppIntroFragment.newInstance(page1));

        SliderPage page2 = new SliderPage();
        page2.setTitle(getString(R.string.intro_navigation));
        page2.setDescription(getString(R.string.intro_navigation_desc));
        page2.setImageDrawable(R.drawable.menu_screen);
        page2.setBgColor(R.color.colorAccent);
        addSlide(AppIntroFragment.newInstance(page2));

        SliderPage page3 = new SliderPage();
        page3.setTitle(getString(R.string.intro_categories));
        page3.setDescription(getString(R.string.intro_categories_desc));
        page3.setImageDrawable(R.drawable.categories_screen);
        page3.setBgColor(R.color.colorAccent);
        addSlide(AppIntroFragment.newInstance(page3));

        SliderPage page4 = new SliderPage();
        page4.setTitle(getString(R.string.intro_search));
        page4.setDescription(getString(R.string.intro_search_desc));
        page4.setImageDrawable(R.drawable.search_screen);
        page4.setBgColor(R.color.colorAccent);
        addSlide(AppIntroFragment.newInstance(page4));

        addSlide(new IntroSupportFragment());
    }

    private void close() {
        Intent intent = new Intent(this, MainActivity.class);
        intent.putExtra(SCROLL_TO_TOP, true);
        startActivity(intent);
        finish();
    }

    @Override
    public void onSkipPressed(Fragment currentFragment) {
        super.onSkipPressed(currentFragment);
        close();
    }

    @Override
    public void onDonePressed(Fragment currentFragment) {
        super.onDonePressed(currentFragment);
        close();
    }
}

File: app/src/main/java/com/commonpepper/photosen/ui/activities/IntroActivity.kt
package com.commonpepper.photosen.ui.activities

import android.content.Intent
import android.os.Bundle
import android.util.TypedValue
import androidx.fragment.app.Fragment
import com.commonpepper.photosen.R.*
import com.commonpepper.photosen.ui.fragments.IntroSupportFragment
import com.github.paolorotolo.appintro.AppIntro
import com.github.paolorotolo.appintro.AppIntroFragment
import com.github.paolorotolo.appintro.model.SliderPage

class IntroActivity : AppIntro() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setGoBackLock(true)
        setDepthAnimation()
        val outValue = TypedValue()
        theme.resolveAttribute(attr.selectableItemBackground, outValue, true)
        skipButton.setBackgroundResource(outValue.resourceId)
        doneButton.setBackgroundResource(outValue.resourceId)
        val page1 = SliderPage()
        page1.apply {
            title = getString(string.welcome)
            description = getString(string.welcome_desc)
            imageDrawable = drawable.big_ic_launcher
            bgColor = color.colorAccent
            addSlide(AppIntroFragment.newInstance(this))
        }
        val page2 = SliderPage()
        page2.apply {
            title = getString(string.intro_navigation)
            description = getString(string.intro_navigation_desc)
            imageDrawable = drawable.menu_screen
            bgColor = color.colorAccent
            addSlide(AppIntroFragment.newInstance(this))
        }
        val page3 = SliderPage()
        page3.apply {
            title = getString(string.intro_categories)
            description = getString(string.intro_categories_desc)
            imageDrawable = drawable.categories_screen
            bgColor = color.colorAccent
            addSlide(AppIntroFragment.newInstance(this))
        }
        val page4 = SliderPage()
        page4.apply {
            title = getString(string.intro_search)
            description = getString(string.intro_search_desc)
            imageDrawable = drawable.search_screen
            bgColor = color.colorAccent
            addSlide(AppIntroFragment.newInstance(this))
        }
        addSlide(IntroSupportFragment())
    }

    private fun close() {
        val intent = Intent(this, MainActivity::class.java)
        intent.putExtra(SCROLL_TO_TOP, true)
        startActivity(intent)
        finish()
    }

    override fun onSkipPressed(currentFragment: Fragment) {
        super.onSkipPressed(currentFragment)
        close()
    }

    override fun onDonePressed(currentFragment: Fragment) {
        super.onDonePressed(currentFragment)
        close()
    }

    companion object {
        const val SCROLL_TO_TOP = "SCROLL_TO_TOP"
    }
}
